src/db/models/status/mod.rs

```diff
@@ -0,0 +1,39 @@
+//! Utility enum representation for the status of a document or library.
+
+/// Enum representation for the status of a document or library.
+pub enum Status {
+    /// 0 - Element added.
+    ElementAdded,
+    /// 1 - Element changed.
+    ElementChanged,
+    /// 2 - Element removed.
+    ElementRemoved,
+    /// 3 - Element effective.
+    ElementEffective,
+}
+
+impl Status {
+    /// Convert a string to a `Status` enum.
+    /// # Errors
+    /// Returns an error if the string is not a valid status value.
+    pub fn from_string(status: &str) -> anyhow::Result<Self> {
+        match status {
+            "Element added" => Ok(Self::ElementAdded),
+            "Element changed" => Ok(Self::ElementChanged),
+            "Element removed" => Ok(Self::ElementRemoved),
+            "Element effective" => Ok(Self::ElementEffective),
+            _ => Err(anyhow::anyhow!("Invalid status value")),
+        }
+    }
+
+    /// Convert a `Status` enum to an integer.
+    #[must_use]
+    pub const fn to_int(&self) -> i64 {
```

Comment:
Very minor, but Postgres supports enums https://www.postgresql.org/docs/current/datatype-enum.html I don't know how easy it is to actually map Rust enums to Postgres enums, but it's something worth bearing in mind. Because seeing say `ElementAdded` when debugging raw SQL output is much nicer than just seeing `0`.